torch/_refs/__init__.py

```diff
@@ -2716,6 +2717,57 @@ def _normalize(
     return out, mean, rstd
 
 
+@register_decomposition(torch.ops.aten.native_group_norm.default, disable_meta=True)
+def native_group_norm(
+    input: Tensor,
+    weight: Optional[Tensor],
+    bias: Optional[Tensor],
+    N: int,
+    C: int,
+    HxW: int,
+    group: int,
+    eps: float,
+) -> Tuple[Tensor, Tensor, Tensor]:
+    utils.check(
+        input.ndim >= 2,
+        lambda: "Expected at least 2 dimensions for input tensor but recieved "
+        + str(input.ndim),
+    )
+    utils.check(
+        C % group == 0,
+        lambda: "Expected number of channels in input to be divisible by num_groups, but got input of shape "
+        + str(input.shape)
+        + " and num_groups="
+        + str(group),
+    )
+    reduction_dims = [2, 3]
+    broadcast_dims = [0] + list(dim for dim in range(2, input.ndim))
+
+    input_reshaped = torch.reshape(input, [N, group, C // group, HxW])
+    out, mean, rstd = _normalize(input_reshaped, reduction_dims, eps)
+    out = out.view(input.shape)
+
+    if weight is None and bias is not None:
+        unsqueeze_bias = prims.expand_dims(bias, broadcast_dims, input.ndim)
+        out = out + unsqueeze_bias
+    elif weight is not None and bias is None:
+        unsqueeze_weight = prims.expand_dims(weight, broadcast_dims, input.ndim)
+        out = out * unsqueeze_weight
+    elif weight is not None and bias is not None:
+        unsqueeze_weight = prims.expand_dims(weight, broadcast_dims, input.ndim)
+        unsqueeze_bias = prims.expand_dims(bias, broadcast_dims, input.ndim)
+        out = out * unsqueeze_weight + unsqueeze_bias
+
+    out = prims.convert_element_type(out, input.dtype)
```

Comment:
Are these actually unconditional conversions, or should this use:

https://github.com/pytorch/pytorch/blob/31931515bc927675136b1637fb9782b9b5ff3174/torch/_prims_common/wrappers.py#L20

?

Reply:
It failed with `AssertionError: Booleans mismatch: True is not False : The torch implementation does not return a view, while the reference does`, so I set `validate_view_consistency=False`